media/bridge.js

```diff
@@ -106,6 +44,20 @@ workspace.addChangeListener(function (event) {
 
 // vscode -> blockly
 function reinstantiateWorkspaceDom(text) {
+	// TODO !!! there might be a better way of using mutations and a common getter/setter block pattern
+	// !!! I really just wanted type checking as it may be needed in Rust generator for conversions
+	// regenerate block definitions so toolbox has custom blocks
+	let temp_dom = Blockly.Xml.textToDom(text);
+	let temp_workspace = visualsolana.workspace_factory();
+	try {
+		Blockly.Xml.domToWorkspace(temp_dom, temp_workspace);
+	} catch (error) {
+		console.log("!!! expected 'undefined block' error as block usage and definition is cyclic");
+		console.log(error);
+		console.log("!!! end expected error");
+	}
+	visualsolana.generate_types_and_instructions(temp_workspace, workspace);
```

Comment:
this is the most problematic part of the diff as a we basically do twice the amount of parsing. However, there is a timeout for reinstantiation due to message publishing on document change so perhaps this isn't as big of a problem. One alternative to catching errors is deleting the whole "interface impl" block though I wasn't able to find the correct `jsdom` APIs for XML.